logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -51,12 +51,12 @@ def fetch_or_store(namespaces, key, default_value = nil)
       # BUT. If the value is not present in the `@fast_lookup` the value will be inserted and we assume that we don't
       # have it in the `@metric_store` for structured search so we add it there too.
 
-      value = @fast_lookup.get(namespaces.dup << key)
+      value = @fast_lookup.get((namespaces.dup << key).hash)
```

Comment:
computing `(namespaces.dup << key)` twice seems odd.

Is namespaces a string? If so, we can just do `(namespaces + key)` and possibly save this to a variable?

The reason I mention is that months from now I wonder if we will look at this line of code and wonder why it was written this way.